### PR TITLE
Node JS - Carlos De La Cruz

### PR DESCRIPTION
**Pull request by CDelacruz1802**
Project by Carlos De La Cruz.
I hope this is correct. Thank you for the time. I liked the sessions 😄 👨‍💻

### CODE REVIEW

File: apps/api/.env
Comment (CDelacruz1802) [inline apps/api/.env:1]: My .gitignore was not ignoring my .env so I deleted the file hehe. If you need it, you can tell me but I know that is not a good practice to have that info in a public way so yeah, I just removed that.

File: apps/api/src/main.ts [hunk 1]
```diff
@@ -1,14 +1,39 @@
...
+app.use(cors(corsOptions));
 
-app.listen(port, host, () => {
-  console.log(`[ ready ] http://${host}:${port}`);
-});
+app.use('/api/categories', routerCatergories);
+app.use('/api/posts', postsRouter);
+app.use('/api/auth', auth);
+
+app.use(verifyToken);
```
Comment (gus-code) [inline apps/api/src/main.ts:26]: If we want to validate the token this need to be before our endpoints, it should look like
```
app.use(verifyToken);

app.use('/api/categories', routerCatergories);
app.use('/api/posts', postsRouter);
app.use('/api/auth', auth);
```

File: apps/api/src/models/category.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import mongoose, { Schema } from 'mongoose';
+import ICategory from '../interfaces/category';
+
+export const categorySchema = new Schema<ICategory>(
+  {
+    id: {
+      type: String,
+      required: [true, 'Property is required']
+    }
```
Comment (gus-code) [inline apps/api/src/models/category.ts:9]: The category is missing the name property here

File: apps/api/src/models/post.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    description: {
+      type: String,
+      required: [true, 'Property is required']
+    },
+    category: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: 'Category'
+    },
+    comments: {
+      type: [{ content: String, author: String }]
```
Comment (gus-code) [inline apps/api/src/models/post.ts:22]: This is a correct approach to have documents nested into another document, to have a better performant we can implement a reference to a different document. Something like this: 

```
comments: [{
  type: Schema.Types.ObjectId,
  ref: 'Comment' // You will need to create a comment model
}]
```

File: package.json [hunk 4]
```diff
@@ -21,6 +21,7 @@
     "axios": "^1.0.0",
     "bcrypt": "^5.1.1",
     "cors": "^2.8.5",
+    "dotenv": "^16.4.1",
```
Comment (gus-code) [inline package.json:24]: This is not needed, Nx automatically import the .env variables into your project

**Pull request merged**